external-crates/move/move-compiler/src/typing/core.rs

```diff
@@ -78,6 +79,13 @@ pub struct Context<'env> {
     pub constraints: Constraints,
 
     loop_info: LoopInfo,
+
+    /// names of modules compiled in this pass (excludes modules coming from pre-compiled libraries)
+    pub compiled_module_names: BTreeSet<ModuleIdent>,
```

Comment:
Not sure I fully understand the filtering here, would it be enough to just check `is_source_module` is true?

Reply:
In other words, I don't think we need this in the context, we can just check at the point of calling `gen_unused_warnings`

Reply:
Yeah, that was dumb...